gcp_variant_transforms/transforms/infer_undefined_headers.py

```diff
@@ -64,9 +62,21 @@ def _get_field_type(self, field_value):
       return vcf_field_conflict_resolver.VcfParserConstants.INTEGER
     elif isinstance(field_value, float):
       return vcf_field_conflict_resolver.VcfParserConstants.FLOAT
+    elif self._can_cast_to(field_value, int):
+      return vcf_field_conflict_resolver.VcfParserConstants.INTEGER
+    elif self._can_cast_to(field_value, float):
+      return vcf_field_conflict_resolver.VcfParserConstants.FLOAT
     else:
       return vcf_field_conflict_resolver.VcfParserConstants.STRING
 
+  def _can_cast_to(self, value, cast_type):
+    """Returns true if `value` can be casted to type `type`"""
+    try:
+      _ = cast_type(value)
+      return True
+    except Exception:
```

Comment:
Please catch `TypeError` instead. Catching generic exceptions are not allowed per styleguide.

Reply:
Catched `ValueError`. Done!